### ISSUE TITLE
Not all KeyValueSelectors get refreshed after sentinel key changed

### ISSUE THREAD

**Issue by lissdy** [Azure/AppConfiguration#953]
Followed https://learn.microsoft.com/en-us/azure/azure-app-configuration/enable-dynamic-configuration-aspnet-core We are using poll model for dynamic configuration 

As the screenshot shows, we set three sentinel keys `service:testService:sentinel`, `shared:testService:sentinel` and `shared:config:sentinel`, and set `RefreshAll` as true for all sentinel keys. 
Three KeyValueSelectors are `shared:testService:*`, `service:testService:*` and `shared:config:*`
Suppose any sentinel value change should trigger the service instance refresh all registered configurations.

![image](https://github.com/user-attachments/assets/b47af5b7-f3ad-4ff1-add2-ec75d89b7303)

But we noticed when we update configuration values and sentinel values, some KeyValueSelectors get refreshed but some are not.  Service still retrieving the old values even 2 days later. We have to restart the service to ensure all configurations are refreshed.

Suppose all selected configuration values should be refreshed in minutes, could you please confirm if the case is normal? How should we ensure selected configuration values get refreshed after configuration change?

SDK version: 6.1.1

**Comment by amerjusupovic** [comment]
Hi @lissdy, by default the refresh interval should be 30 seconds if you haven't set it manually in `ConfigureRefresh`. I'm not able to reproduce this behavior after copying the quickstart locally with version 6.1.1 of `Microsoft.Azure.AppConfiguration.AspNetCore`. I selected multiple key-value collections as you did and set a few different sentinel key-values. 

Here are some things I'd want to check from your end:

1. Are you refreshing your application after at least one sentinel key-value and any selected key-values are updated in your store (you may need to refresh a couple times)?
2. Have you made all calls to `AddAzureAppConfiguration` and `UseAzureAppConfiguration` in the order shown in the quickstart?
3. If you can enable logging at the `Debug` level, can you check your logs to see which key-values are changing and if the sentinel key-value is logged as a change? The instructions for this can be found in the page you sent.

**Comment by amerjusupovic** [comment]
You're right, if a key is registered with `refreshAll` set to true then all selected key-values should be refreshed. Is there any way you could try upgrading to the newest provider version, 7.3.0?
